src/library_fs.js

```diff
@@ -664,8 +667,8 @@ FS.staticInit();
     mknod(path, mode, dev) {
       var lookup = FS.lookupPath(path, { parent: true });
       var parent = lookup.node;
-      var name = PATH.basename(path);
-      if (!name || name === '.' || name === '..') {
+      var name = path.split('/').filter((p) => !!p && (p !== '.')).pop();
```

Comment:
I would hope that PATH.basename does something very much like this?  If `PATH.basename` doesn't to the right thing should we fix it?

Also is `mknod("/doo/bar/.")` actually supposed to work?  Seems like it shouldn't

Reply:
It looks like node's path.basename doesn't call normalize so likely ours should not either:

```
> path.basename("foo/bar/../a/b/..")
'..'
```

Reply:
Sounds good to me.

Reply:
Perhaps I should add a test that `mknod("a/.")` raises an error while I'm at it.